solver/src/driver.rs

```diff
@@ -125,38 +126,89 @@ impl Driver {
         }
     }
 
-    // Returns only settlements for which simulation succeeded.
+    // Split settlements into successfully simulating ones and errors.
     async fn simulate_settlements(
         &self,
         settlements: Vec<RatedSettlement>,
-    ) -> Result<Vec<RatedSettlement>> {
+    ) -> Result<(Vec<RatedSettlement>, Vec<(RatedSettlement, Error)>)> {
         let simulations = settlement_simulation::simulate_settlements(
             settlements
                 .iter()
                 .map(|settlement| settlement.settlement.clone().into()),
             &self.settlement_contract,
             &self.web3,
             &self.network_id,
+            settlement_simulation::Block::LatestWithoutTenderly,
         )
         .await
         .context("failed to simulate settlements")?;
+
         Ok(settlements
             .into_iter()
             .zip(simulations)
-            .filter_map(|(settlement, simulation)| match simulation {
-                Ok(()) => Some(settlement),
-                Err(err) => {
-                    tracing::error!("settlement simulation failed\n error: {:?}", err);
-                    tracing::debug!("settlement failure for: \n{:#?}", settlement.settlement,);
-                    None
-                }
-            })
-            .collect())
+            .partition_map(|(settlement, result)| match result {
+                Ok(()) => Either::Left(settlement),
+                Err(err) => Either::Right((settlement, err)),
+            }))
+    }
+
+    // Log simulation errors only if the simulation also fails in the block at which on chain
+    // liquidity was queried. If the simulation succeeds at the previous block then the solver
+    // worked correctly and the error doesn't have to be reported.
+    // Note that we could still report a false positive because the earlier block might be off by if
+    // the block has changed just as were were querying the node.
+    async fn report_simulation_errors(
+        &self,
+        errors: Vec<(RatedSettlement, Error)>,
+        current_block_during_liquidity_fetch: u64,
+    ) {
+        let simulations = match settlement_simulation::simulate_settlements(
+            errors
+                .iter()
+                .map(|(settlement, _)| settlement.settlement.clone().into()),
+            &self.settlement_contract,
+            &self.web3,
+            &self.network_id,
+            settlement_simulation::Block::FixedWithTenderly(current_block_during_liquidity_fetch),
+        )
+        .await
+        {
+            Ok(simulations) => simulations,
+            Err(err) => {
+                tracing::error!(
```

Comment:
When does this error happen? I initially confused this for "simulation failed" but it's more of a meta level error isn't it? Maybe we can make it a bit clearer in the error message.